routes/api/ads.js

```diff
@@ -0,0 +1,89 @@
+const { Router } = require('express');
+const { Op } = require('sequelize');
+const { ad } = require('../../databases/db');
+
+const adsRouter = Router();
+const allowedFields = ['announcer', 'adType', 'redirecTo', 'img'];
+
+const httpGetResponse = (res, resource, resourceName) => {
+  if (resource) {
+    res.status(200).json(resource);
+  } else {
+    res.status(404).send(resourceName + ' not found');
+  }
+}
+
+const findAdById = (id) => {
+  return ad.findByPk(id);
+}
+
+adsRouter.get('/', async (req, res) => {
+  const ads = await ad.findAll();
+  httpGetResponse(res, ads, 'ads');
+});
+
+adsRouter.get('/search', async (req, res) => {
```

Comment:
Creo que estas son las que manejamos, para la lista neta "/all" (que no se si haya falta lista neta en ads 🤔 ) y.... para la paginacion normal el nombre del recurso y listo

```suggestion
adsRouter.get('/all', async (req, res) => {
  const ads = await ad.findAll();
  httpGetResponse(res, ads, 'ads');
});

adsRouter.get('/', async (req, res) => {
```